### ISSUE TITLE
Uncaught Error: Could not find module `ember-cli-cordova-shims/services/notifications`

### ISSUE THREAD

**Issue by shadowbrush** [Skalar/ember-cli-cordova-shims#4]
We're trying to get push notifications to work in our Ember/Cordova project. Following the instructions in your readme, we installed the two cordova plug-in dependencies (device, PushPlugin), and ran `npm i ember-cli-cordova-shims`. We added the initializer, but the import fails with: `Uncaught Error: Could not find module 'ember-cli-cordova-shims/services/notifications'`. We do have a `/node_modules/ember-cli-cordova-shims`. Thanks!

**Comment by theodorton** [comment]
I'll look into this tomorrow. Thanks for your feedback :)

**Comment by theodorton** [comment]
Are you on v0.3.0?
Also could you try the following instead and see how that works?

``` javascript
import NotificationsService from "{your-app}/services/notifications";
...
```

**Comment by rocketbop** [comment]
Hi Theo, I have the same problem as parent. :/ Using version 3.0.0

I did:

cd node_modules/
git clone git@github.com:Skalar/ember-cli-cordova-shims.git
cd ember-cli-cordova-shims
npm install
bower install

But the import cannot be found, exactly the same as above. I also tried;

```
import NotificationsService from 'ember-cli-cordova-shims/app/services/notifications';
```

Uncaught Error: Could not find module `ember-cli-cordova-shims/services/notifications` imported from `my-app/initializers/pushplugin`

EDIT: My issue seems to have just gone away. :8ball:

**Comment by shadowbrush** [comment]
Hi, thanks for responding so quickly! Did you make any changes? It is now working, with the import line from the readme! 

Over the weekend, when I tried to add ember-cli-cordova-shims to package.json, I received an error message from npm. That error message is now gone. 

Thanks again!

Holger

**Comment by shadowbrush** [comment]
Hi Theo(?), 

I've run into another issue. Following your readme instructions, I've added the service to the Ember application routes, and hooked up a button with the `subscribeForNotifications` action. However, this line in `routes/application.js`:

```
    this.notifications.register().then(function (data) {
```

raises the following exception:

```
    Uncaught TypeError: this.notifications.register is not a function
```

The object `this.notifications` is set to `function members@service:notifications`. Any ideas? 

I never answered your earlier question: yes, I am using version 0.3.0. 

Thanks!

Holger

**Comment by rocketbop** [comment]
Yep, I have the same issue. I debugged, and in the console, and this.notifications is a function.

**Comment by theodorton** [comment]
You should use `this.get('notifications').register...` :)

**Comment by rocketbop** [comment]
That doesn't seem to work, still. I get:

Uncaught TypeError: this.get(...).register is not a function

:octopus:

**Comment by shadowbrush** [comment]
I'm new to Ember services, and don't know what to expect from the "notifications" objects, but here are the "own" methods available from the object: 

willReopen,name,_lazyInjections,isMethod,extend,length,ClassMixin,__super__,_onLookup,detect,eachComputedProperty,_computedProperties,caller,proto,createWithMixins,reopen,_toString,isClass,detectInstance,_initMixins,reopenClass,isServiceFactory,superclass,toString,__ember_meta__,**ember_listens**,_initProperties,_super,_debugContainerKey,PrototypeMixin,arguments,prototype,metaForProperty,create

**Comment by shadowbrush** [comment]
I think it may have been an Ember version conflict (I'm on 1.13.1). I moved the service to /app/services, removed the initializer, some minor changes, and it's working now. Paul, if you want my code, ping me. Theo, thanks for you help!

Holger

**Comment by rocketbop** [comment]
Hey thanks Shadowbrush, that would be amazing! I stopped trying to get it to work last week as i was working on some other parts of my appl, but would love to give it another go :)
